### ISSUE TITLE
Parser error for valid Java code

### ISSUE THREAD

**Issue by jpschewe** [typetools/checker-framework#4614]
I'm using checker 3.12.0 with OpenJDK 11.

This code
```java
        final String versionInfo = Version.getAllVersionInformation().entrySet().stream() //
                                          .map(e -> String.format("%s:%s", e.getKey(), e.getValue())) //
                                          .collect(Collectors.joining("\n"));
```

Results in the parser error
```
warning: [expression.unparsable.type.invalid] Expression invalid in dependent type annotation: [error for expression: ()java.util.Map<java.lang.String,java.lang.String>.getAllVersionInformation(); error: Invalid '()java.util.Map<java.lang.String,java.lang.String>.getAllVersionInformation()' because the expression did not parse. Error message: Encountered unexpected token: "java" <IDENTIFIER>]
/home/jpschewe/projects/fll-sw/working-dir/src/main/java/fll/web/GatherBugReport.java:91: warning: [lambda.param.type.incompatible] incompatible parameter types for parameter 0 in lambda expression.
                                          .map(e -> String.format("%s:%s", e.getKey(), e.getValue())) //
                                               ^
  found   : Entry<@KeyFor("()java.util.Map<java.lang.String,java.lang.String>.getAllVersionInformation()") String, String>
  required: Entry<@KeyFor("[error for expression: ()java.util.Map<java.lang.String,java.lang.String>.getAllVersionInformation(); error: Invalid '()java.util.Map<java.lang.String,java.lang.String>.getAllVersionInformation()' because the expression did not parse. Error message: Encountered unexpected token: "java" <IDENTIFIER>]") String, String>
```

**Comment by jpschewe** [comment]
Stand-alone example.

```java
package net.mtu.eggplant.checker.parser_error;

import java.io.IOException;
import java.io.InputStream;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import java.util.Properties;
import java.util.stream.Collectors;

/**
 * Get version information about the FLL software.
 *
 * @author jpschewe
 */
public final class Version {

  private static final Map<String, String> VERSION_INFORMATION;

  private static final String VERSION;
  static {
    final InputStream stream = Version.class.getResourceAsStream("/fll/resources/git.properties");
    if (null == stream) {
      System.out.println("Unable to find version.properties!");
      VERSION = "NO-PROPERTIES-FILE";
      VERSION_INFORMATION = Collections.emptyMap();
    } else {
      final Properties versionProps = new Properties();
      String version;
      try {
        versionProps.load(stream);
        version = versionProps.getProperty("git.build.version", "NO-PROPERTY");
      } catch (final IOException ioe) {
        System.out.println("Error loading version properties");
        ioe.printStackTrace();
        version = "EXCEPTION";
      } finally {
        try {
          stream.close();
        } catch (final IOException e) {
          System.out.println("Error closing stream");
          e.printStackTrace();
        }
      }

      // convert to String, String
      final Map<String, String> map = new HashMap<>();
      for (final String name : versionProps.stringPropertyNames()) {
        map.put(name, versionProps.getProperty(name));
      }

      VERSION_INFORMATION = Collections.unmodifiableMap(map);
      VERSION = version;
    }
  }

  private Version() {
    // no instances
  }

  /**
   * Get the version of the software being used.
   *
   * @return the version
   */
  public static String getVersion() {
    if (VERSION.contains("APP-VERSION")) {
      return "devel";
    } else {
      return VERSION;
    }
  }

  /**
   * @return all version information stored in version.properties, unmodifiable
   *         map
   */
  public static Map<String, String> getAllVersionInformation() {
    return VERSION_INFORMATION;
  }

  public void method() {
    final String versionInfo = Version.getAllVersionInformation().entrySet().stream() //
      .map(e -> String.format("%s:%s", e.getKey(), e.getValue())) //
      .collect(Collectors.joining("\n"));
  }
  
}
```

Error
```
/home/jpschewe/projects/checker-bugs/src/main/java/net/mtu/eggplant/checker/parser_error/Version.java:84: error: [lambda.param.type.incompatible] incompatible parameter types for parameter 0 in lambda expression.
      .map(e -> String.format("%s:%s", e.getKey(), e.getValue())) //
           ^
  found   : Entry<@KeyFor("()java.util.Map<java.lang.String,java.lang.String>.getAllVersionInformation()") String, String>
  required: Entry<@KeyFor("[error for expression: ()java.util.Map<java.lang.String,java.lang.String>.getAllVersionInformation(); error: Invalid '()java.util.Map<java.lang.String,java.lang.String>.getAllVersionInformation()' because the expression did not parse. Error message: Encountered unexpected token: "java" <IDENTIFIER>]") String, String>
```

**Comment by mernst** [comment]
Thanks for the full reproduction.  That helps us a lot.  Here is a minimized version of your test case:

```java
import java.util.HashMap;
import java.util.Map;
import java.util.stream.Collectors;

public final class Issue4614 {

  public static Map<String, String> getAllVersionInformation() {
    return new HashMap<>();
  }

  public void method1() {
    final String versionInfo =
        Issue4614.getAllVersionInformation().entrySet().stream() //
            .map(e -> String.format("%s:%s", e.getKey(), e.getValue())) //
            .collect(Collectors.joining("\n"));
  }

  Map<String, String> allVersionInformation = new HashMap<>();

  public void method2() {
    final String versionInfo =
        allVersionInformation.entrySet().stream() //
            .map(e -> String.format("%s:%s", e.getKey(), e.getValue())) //
            .collect(Collectors.joining("\n"));
  }
}
```

The output is:
```
javacheck -g Issue4614.java -processor nullness
error: [expression.unparsable] Expression invalid in dependent type annotation: [error for expression: ()java.util.Map<java.lang.String,java.lang.String>.getAllVersionInformation(); error: Invalid '()java.util.Map<java.lang.String,java.lang.String>.getAllVersionInformation()' because the expression did not parse. Error message: Encountered unexpected token: "java" <IDENTIFIER>]
Issue4614.java:14: error: [lambda.param] incompatible parameter types for parameter 0 in lambda expression.
            .map(e -> String.format("%s:%s", e.getKey(), e.getValue())) //
                 ^
  found   : Entry<@KeyFor("()java.util.Map<java.lang.String,java.lang.String>.getAllVersionInformation()") String, String>
  required: Entry<@KeyFor("[error for expression: ()java.util.Map<java.lang.String,java.lang.String>.getAllVersionInformation(); error: Invalid '()java.util.Map<java.lang.String,java.lang.String>.getAllVersionInformation()' because the expression did not parse. Error message: Encountered unexpected token: "java" <IDENTIFIER>]") String, String>
2 errors
```

Note that there is no error when using a field, only when calling a method.
